08 - Fun with HTML5 Canvas/index-CLASS.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>HTML5 Canvas</title>
+</head>
+  <body>
+    <canvas id="draw" width="800" height="800"></canvas>
+    <button class="clearCanvas">Clear Canvas</button>
+    <script>
+        //// classes
+        class DrawingCanvas {
+            constructor(id, contextID = '2d') {
+                this.canvas = document.querySelector(id);
+                this.context = this.canvas.getContext(contextID)
+                this.X = 0;
+                this.Y = 0;
+                this.hue = 0;
+                this.isDrawing = false;
+                this.isStrokeGrowing = true;
+
+                //// binding this to events
+                this.startDrawing = this.startDrawing.bind(this);
+                this.stopDrawing = this.stopDrawing.bind(this);
+                this.clearCanvas = this.clearCanvas.bind(this);
+                this.draw = this.draw.bind(this);
+            }
+
+            setCanvas() {
```

Comment:
let's think about classes like a service you are providing to other developers. you make up the "contract" how to use it and others can do so.

here: why is this an extra method which the "developer" has to (remember to) call? why is this not part of the constructor? does it make sense to not call this for the developer?

Reply:
sure! this extra method does not actually provide any added-value. I removed it!
In a more complex class we could maybe create a method to setFatLine() or to set a particular stroke (just stupid examples) where many different settings could be called with one method instead of passing too many arguments to the constructor.